doc/nrf/ug_thingy91.rst

```diff
@@ -0,0 +1,262 @@
+
+There are multiple methods of programming a sample or application onto a Thingy:91.
+You can choose the method based on the availability or absence of an external debug probe to program.
+
+.. note::
+
+   In the event of not having an external debug probe to program the Thingy:91, you can directly program by :ref:`using the USB (MCUboot) method and nRF Connect Programmer<programming_usb>`.
+   The ``app_signed.hex`` firmware image file is used for the update in this scenario.
+
+.. _build_pgm_segger:
+
+Building and programming using SEGGER Embedded Studio
+=====================================================
+
+.. include:: gs_programming.rst
+   :start-after: build_SES_projimport_open_start
+   :end-before: build_SES_projimport_open_end
+
+
+.. figure:: images/ses_thingy_configuration.png
+   :alt: Opening the AT Client sample
+
+   Opening the AT Client sample for the nrf9160_pca20035ns build target
+
+.. note::
+
+   The *Board Directory* folder can be found in the following location: ``ncs/nrf/boards/arm``.
+
+
+4. Click **OK** to import the project into SES. You can now work with the project in the IDE.
+
+   .. include:: gs_programming.rst
+      :start-after: build_SES_projimport_note_start
+      :end-before: build_SES_projimport_note_end
+
+#. To build the sample or an application:
+
+   a. Select your project in the Project Explorer.
+   #. From the menu, select **Build -> Build Solution**.
+      This builds the project.
+
+   You can find the output of the build, which includes the merged HEX file containing both the application and the SPM, in the ``zephyr`` sub-folder in the build directory.
+
+.. prog_extdebugprobe_start
+
+6. To program the sample or application:
+
+   a. Set the Thingy:91 SWD selection switch (**SW2**) to **nRF91** or **nRF52** depending on whether you have to program the nRF9160 SiP or the nRF52840 SoC component.
+   #. Connect the Thingy:91 to the debug out port on a 10-pin external debug probe, for example, nRF9160 DK (Development Kit), using a 10-pin JTAG cable.
+
+      .. note::
+
+         If you are using nRF9160 DK as the debug probe, make sure that **VDD_IO (SW11)** is set to 1.8 V on the nRF9160 DK.
+
+   #. Connect the external debug probe to the PC using USB.
+   #. Make sure that the Thingy:91 and the external debug probe are powered on.
+
+.. prog_extdebugprobe_end
+..
+
+   e. In SES, select **Target -> Connect J-Link**.
+   #. Select **Target -> Download zephyr/merged.hex** to program the sample or application onto Thingy:91.
+   #. The device will reset and run the programmed sample or application.
+
+.. _build_pgm_cmdline:
+
+Building and programming using Command line
+===========================================
+
+Complete the command line build setup (see the instructions for :ref:`Windows<build_environment_cli_win>` and :ref:`Linux<build_environment_cli_linux>`) before you start building |NCS| projects on the command line.
```

Comment:
Do we have instructions for macOS as well?

Reply:
Yes, https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/gs_ins_mac.html

Reply:
Added this.